input/input.go

```diff
@@ -111,7 +112,7 @@ func (i *inputKit) RequiredTxOut() *wire.TxOut {
 // must be used in the transaction including it. This will be false for the
 // base input type since we can re-sign for any lock time.
 func (i *inputKit) RequiredLockTime() (uint32, bool) {
```

Comment:
Why modify this _default_ version instead of making a new concrete input type to expose this additional functionality?

Reply:
I decided not to based on the following:

```
// BaseInput contains all the information needed to sweep a basic output
// (CSV/CLTV/no time lock)
type BaseInput struct {
	inputKit
}
```